### PR TITLE
Fix checkout.ThankYouView.get_object to never raise DoesNotExist error

### PR DESCRIPTION
**Pull request by jwayodi**
Also fix "is_anonymous" attribute value in WebTestCase subclass.

### CODE REVIEW

File: src/oscar/apps/checkout/views.py [hunk 1]
```diff
@@ -672,16 +672,16 @@ def get_object(self, queryset=None):
...
             if 'order_number' in self.request.GET:
-                order = Order._default_manager.get(
-                    number=self.request.GET['order_number'])
+                order = Order._default_manager.filter(
+                    number=self.request.GET['order_number']).first()
             elif 'order_id' in self.request.GET:
-                order = Order._default_manager.get(
-                    id=self.request.GET['order_id'])
+                order = Order._default_manager.filter(
+                    id=self.request.GET['order_id']).first()
```
Comment (sasha0) [inline src/oscar/apps/checkout/views.py:679]: Can be a bit simpler:

```python
kwargs = {}
if 'order_number' in self.request.GET:
    kwargs['number'] = self.request.GET['order_number']}
elif 'order_id' in self.request.GET:
    kwargs['id'] = self.request.GET['order_id']).first()
order = Order._default_manager.filter(**kwargs).first()
```

**Review by sasha0 — CHANGES_REQUESTED** [review]

File: src/oscar/apps/checkout/views.py (hunk 1 above)
Comment (jwayodi) [reply src/oscar/apps/checkout/views.py]: Thanks. Fixed.

File: tests/integration/checkout/test_views.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 
 class ThankYouViewTestCase(WebTestCase):
-    is_anonymous = False
+    is_anonymous = True
```
Comment (sasha0) [inline tests/integration/checkout/test_views.py:8]: Hm, aren't existing test methods supposed to be run for anonymous users?

Comment (jwayodi) [reply tests/integration/checkout/test_views.py:8]: Yes, the test methods all use `OSCAR_ALLOW_ANON_CHECKOUT = True`, and make anonymous (without specifying a user) requests. So I am not sure why `is_anonymous` would be `False`?

**Comment by jwayodi** [comment]
@sasha0 I have fixed the tests to properly use Django's test client to make requests after modifying the session. I was previously using WebTest to make these requests, which was incorrect.

Also, `tests.integration.checkout.test_views.ThankYouViewTestCase.test_missing_order_id_in_the_session` looks similar to `tests.functional.checkout.test_customer_checkout.TestThankYouView.tests_gets_a_302_when_there_is_no_order`, only that the former makes an anonymous request. I added `tests.integration.checkout.test_views.ThankYouViewTestCase.test_order_id_in_the_session_is_for_a_non_existent_order` in addition to `tests.functional.checkout.test_customer_checkout.TestThankYouView.tests_gets_a_302_when_the_order_id_in_the_session_is_for_a_non_existent_order`, to match the previous case.

**Review by solarissmoke — APPROVED** [review]

**Pull request merged**